Scripts/Interfaces/Entity/IMenu.cs

```diff
@@ -22,6 +22,11 @@ public interface IMenu
 		/// Root GameObject for visible menu content
 		/// </summary>
 		GameObject menuContent { get; }
+
+		/// <summary>
+		/// If the rayOrigin this menu is attached to is hovering UI, hide it if the raycast distance is less than this
+		/// </summary>
+		float hideDistance { get; }
```

Comment:
Hm. Since I only needed a distance along the ray, I kept it to a single float. I was also not a huge fan of something so implementation-specific. Exposing localbounds is a good idea, though it does still contain the expectation that they're local to "something."

Something else to consider is whether this belongs in public interfaces or if we want to handle our "special case" main menu that's attached to your hand separately from other menu features.

The concept of auto-hiding the menu when it intersects with things or an associated ray is hovering UI is conceptually coupled to the idea of "menu on rayorigin," so I figured it was safe to assume at this level.

TL;DR: I think just exposing localbounds makes sense in the scope of this PR.